cmd/agent/main.go

```diff
@@ -185,31 +187,70 @@ func enableEdgeMode(tunnelOperator agent.TunnelOperator, clusterService agent.Cl
 				return err
 			}
 		}
+	} else {
+		log.Println("[DEBUG] [main,edge] [message: Edge key not specified. Serving Edge UI]")
+		edgeServer := http.NewEdgeServer(tunnelOperator, clusterService)
 
-		return tunnelOperator.Start()
-	}
+		go func() {
+			log.Printf("[INFO] [main,edge,http] [server_address: %s] [server_port: %s] [message: Starting Edge server]", options.EdgeServerAddr, options.EdgeServerPort)
 
-	log.Println("[DEBUG] [main,edge] [message: Edge key not specified. Serving Edge UI]")
-	edgeServer := http.NewEdgeServer(tunnelOperator, clusterService)
+			err := edgeServer.Start(options.EdgeServerAddr, options.EdgeServerPort)
+			if err != nil {
+				log.Fatalf("[ERROR] [main,edge,http] [message: Unable to start Edge server] [error: %s]", err)
+			}
 
-	go func() {
-		log.Printf("[INFO] [main,edge,http] [server_address: %s] [server_port: %s] [message: Starting Edge server]", options.EdgeServerAddr, options.EdgeServerPort)
+			log.Println("[INFO] [main,edge,http] [message: Edge server shutdown]")
+		}()
 
-		err := edgeServer.Start(options.EdgeServerAddr, options.EdgeServerPort)
-		if err != nil {
-			log.Fatalf("[ERROR] [main,edge,http] [message: Unable to start Edge server] [error: %s]", err)
-		}
+		go func() {
+			timer1 := time.NewTimer(agent.DefaultEdgeSecurityShutdown * time.Minute)
+			<-timer1.C
 
-		log.Println("[INFO] [main,edge,http] [message: Edge server shutdown]")
-	}()
+			if !tunnelOperator.IsKeySet() {
+				log.Printf("[INFO] [main,edge,http] [message: Shutting down Edge UI server as no key was specified after %d minutes]", agent.DefaultEdgeSecurityShutdown)
+				edgeServer.Shutdown()
+			}
+		}()
+	}
 
-	go func() {
-		timer1 := time.NewTimer(agent.DefaultEdgeSecurityShutdown * time.Minute)
-		<-timer1.C
+	loopIntervalFrequency, err := time.ParseDuration(agent.DefaultConfigCheckInterval)
+	if err != nil {
+		return err
+	}
 
-		if !tunnelOperator.IsKeySet() {
-			log.Printf("[INFO] [main,edge,http] [message: Shutting down Edge UI server as no key was specified after %d minutes]", agent.DefaultEdgeSecurityShutdown)
-			edgeServer.Shutdown()
+	ticker := time.NewTicker(time.Duration(loopIntervalFrequency))
+
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				log.Printf("[DEBUG] [main,edge,poll] [message: checking docker config and key]")
+
+				key := tunnelOperator.GetKey()
+				if key == "" {
+					continue
+				}
+
+				agentTags, err := infoService.GetInformationFromDockerEngine()
+				if err != nil {
+					log.Printf("[ERROR] [main,edge,poll] [message: an error occured during docker config check] [error: %s]", err)
+					continue
+				}
+
```

Comment:
Add the following DEBUG statement here:

```
log.Printf("[DEBUG] [main,edge,docker] [] [message: Docker runtime configuration check] [engine_status: %s] [leader_node: %s]", agentTags[agent.MemberTagEngineStatus], agentTags[agent.MemberTagKeyIsLeader] == "1")
```

See comment above and below about removing other DEBUG statements.